website/views/shared/tasks/task_view/skills.jade

```diff
@@ -8,16 +8,26 @@ ul.items.rewards
 
 // Class Skills
 ul.items(ng-if='main && list.type=="reward" && user.stats.class && !user.preferences.disableClasses')
-  li.task.reward-item(ng-repeat='(k,skill) in Content.spells[user.stats.class]', ng-if='user.stats.lvl >= skill.lvl',popover-trigger='mouseenter', popover-placement='top', popover='{{skill.notes()}}')
+  li.task.reward-item.list-group-item(ng-repeat='(k,skill) in Content.spells[user.stats.class]', ng-if='user.stats.lvl >= skill.lvl',popover-trigger='mouseenter', popover-placement='top',
+  popover='{{skill.notes()}}{{k=="frost" && user.stats.buffs.streaks ? " You have already cast this today. Your streaks are frozen, and there\'s no need to cast this again." : ""}}}{{k=="stealth" ? " Number of dailies avoided: " + user.stats.buffs.stealth + "." : ""}}{{k=="stealth" && user.stats.buffs.stealth >= user.dailys.length ? " You have already avoided all your dailies, there\'s no need to cast this again." : ""}}')
     .task-meta-controls
       span.task-notes
         span.glyphicon.glyphicon-comment
     //left-hand size commands
     .task-controls.task-primary
-      a.money.btn-buy.item-btn(ng-click='castStart(skill)', ng-class='{active: skill.key == spell.key}')
+      a.money.btn-buy.item-btn(ng-click='castStart(skill)', ng-if='!( k=="frost" || k=="stealth" )')
+        span.reward-cost
+          strong {{::skill.mana}}
+          =env.t('mp')
+      a.btn.money.btn-buy.item-btn(ng-click='castStart(skill)', ng-class='{"disabled": user.stats.buffs.streaks}', ng-if='k=="frost"')
```

Comment:
Instead of repeating this three times, you could just make a function that determines if the disabled class should be applied or not. IE:

```js
// if is frost and streaks buff is applied - disabled
// if is stealth and stealth buff greater than dailies - disabled
// esle not disabled
```